dask/core.py

```diff
@@ -218,9 +218,26 @@ def iskey(key: object) -> bool:
 
 
 def validate_key(key: object) -> None:
-    """Validate the format of a dask key."""
-    if not iskey(key):
-        raise TypeError(f"Unexpected key type {type(key)} (value: {key!r})")
+    """Validate the format of a dask key.
+
+    See Also
+    --------
+    iskey
+    """
+    if iskey(key):
+        return
+    typ = type(key)
+
+    if typ is tuple:
+        _index = None
+        try:
+            for _index, part in enumerate(cast(tuple, key)):
+                validate_key(part)
+        except TypeError as e:
+            raise TypeError(
+                f"Composite key contains unexpected key type at {_index=} ({key=!r})"
```

Comment:
```suggestion
                f"Composite key contains unexpected key type at index={_index} ({key=!r})"
```

That would get ci passing

Reply:
I shouldn't have followed `flake8`s advice there. Reverted to `index` and ignoring the check.

Reply:
Sounds good to me